compiler/vm/vmincr.nim

```diff
@@ -0,0 +1,189 @@
+## This module implements the bridging parts required for execution
+## interleaved with lazy code generation. In interleaved mode, only the
+## immediately required functions are ran through code-generation.
+##
+## This works as follows:
+## - a not yet generated function is encountered -> invoke `vmgen`
+## - the code is generated and new dependcies are collected
+## - the collected new dependencies are loaded into the execution environment
+## - the function table entry is updated with the bytecode offset and the
+##  required register count
+## - execution resumes
+
+# TODO: the module needs a better name
```

Comment:
I'd suggest:
- vmgenincremental
- vmjit
- vmgenasyougo

It's jit in my opinion, I added a bunch because I'm silly. :D

We're generating the least amount as we need it, just in time.

Reply:
`vmjit` fits really good, thanks. "just-in-time code-generation" is also much more descriptive than "incremental execution"